corehq/apps/app_manager/tests/test_views.py

```diff
@@ -88,3 +99,92 @@ def test_edit_commcare_profile(self):
         custom_properties = content["changed"]["custom_properties"]
 
         self.assertEqual(custom_properties["random"], "changed")
+
+    def _load_app(self, filename):
+        with open(os.path.join(os.path.dirname(__file__), 'data', filename)) as f:
+            self.app = import_app(json.loads(f.read()), self.domain)
+            self.app.build_spec = BuildSpec.from_string('2.7.0/latest')
+            self.build = self.app.make_build()
+            self.build.save()
+
+    def _test_status_codes(self, names, kwargs):
+        for name in names:
+            response = self.client.get(reverse(name, kwargs=kwargs), follow=False)
+            self.assertEqual(response.status_code, 200)
+
+    def test_basic_app(self):
+        self._load_app('basic_app.json')
+        self._test_status_codes([
+            'view_app',
+            'release_manager',
+            AppSummaryView.urlname,
+        ], {
+            'domain': self.domain,
+            'app_id': self.app.id,
+        })
+
+    def test_user_registration(self):
+        self._load_app('user_registration.json')
+        self._test_status_codes([
+            'view_user_registration',
+            'user_registration_source',
+        ], {
+            'domain': self.domain,
+            'app_id': self.app.id,
+        })
+
+    def test_module(self):
+        self._load_app('basic_app.json')
+        self._test_status_codes(['view_module'], {
+            'domain': self.domain,
+            'app_id': self.app.id,
+            'module_id': 0,
+        })
+
+    def test_advanced_module(self):
+        self._load_app('advanced_module.json')
+        self._test_status_codes(['view_module'], {
+            'domain': self.domain,
+            'app_id': self.app.id,
+            'module_id': 0,
+        })
+
+    def test_reporting_module(self):
+        self._load_app('reporting_module.json')
+        self._test_status_codes(['view_module'], {
+            'domain': self.domain,
+            'app_id': self.app.id,
+            'module_id': 0,
+        })
+
+    def test_view_form(self):
+        self._load_app('basic_app.json')
```

Comment:
(doesn't really concern this PR, but more of a general question about tests) For the ease of readability, explicitness and less changes in the app-source when the App-Model changes, one could argue to create the application on the fly using methods like `new_app`, and explicitly making it what you want using helper methods - but that could slow down the tests.

Wherever possible I prefer creating apps using the Application API methods and making it what you want vs loading it from a source file.
- Whenever app-model changes slightly, you would have to change all these source files - with API methods, you probably would have to change just those public methods
- Using API methods makes it explicit to the reader what a test expects the app be like.

But on the other end, loading from json files and avoiding database makes these tests fast. So I am not really sure what to choose, but I usually try to go with API methods. I am curious to listen what you and others think.